Announcing FusionAuth 1.51 (#3062)
FusionAuth/fusionauth-site: astro/src/content/blog/announcing-fusionauth-1-51.mdx

---
publish_date: 2024-05-30
title: Announcing FusionAuth 1.51 - Theming Toucan
description: This update includes our new visual, Simple Theme editor and a couple bug fixes
authors: Alex Patterson
image: /img/blogs/release-1-51/fusionauth-1-51.png
categories: Release Updates, Product
tags: release notes, themes
excerpt_separator: "{/* more */}"
---
import { YouTube } from '@astro-community/astro-embed-youtube';
import Breadcrumb from 'src/components/Breadcrumb.astro';
import InlineUIElement from 'src/components/InlineUIElement.astro';

> FusionAuth versions 1.51.0 shipped in late May, 2024. This version includes our new visual, Simple Theme editor and a couple bug fixes.

Reviewer: ```suggestion
FusionAuth version 1.51.0 shipped in late May, 2024. This version includes our new visual, Simple Theme editor and a couple bug fixes.
```

{/* more */}

> Something about a [Toucan Can-Can](https://www.youtube.com/shorts/5Uq0MUiQwLE), so this release shall be the "Theming Toucan".

Reviewer 2: Love the video, but "Something about a Toucan Can-Can" feels like an incomplete thought.  Did you mean to add something here like Tocan's are unique and your themes should be too...".

> Something about a [Toucan Can-Can](https://www.youtube.com/shorts/5Uq0MUiQwLE), so this release shall be the "Theming Toucan".

Author: ```suggestion
[Toucans are fun and unique](https://www.youtube.com/shorts/5Uq0MUiQwLE) and your themes should be too, so this release shall be the "Theming Toucan".
```

![Toucan - Would you like to simplify your theme?](/img/blogs/release-1-51/toucan.jpg)

# FusionAuth 1.51: The Theming Toucan Release

> Welcome to the FusionAuth 1.51 "Theming Toucan" release post! In this post, we'll delve into the details of the newly introduced visual, Simple Theme editor that revolutionizes how you can customize your login forms. We'll explore how this release simplifies the theming process, compare it to the Advanced way of theming, and provide you with a detailed walkthrough to help you get the most out of this powerful feature.

Author: ```suggestion
Welcome to the FusionAuth 1.51 "Theming Toucan" release post! In this post, we'll delve into the details of the newly introduced visual, Simple Theme editor that revolutionizes how you can customize your login forms. We'll explore how this release simplifies the theming process, compare it to the Advanced way of theming, and provide you with a detailed walk through to help you get the most out of this powerful feature.
```

TL;DR - Just watch this video, since non-coders don't really read docs anyways 😉.

<YouTube id="MZypd5RUAzs" />

## Table of Contents

1. [Introduction to FusionAuth](#introduction-to-fusionauth)
1. [Overview of the Theming Toucan Release](#overview-of-the-theming-toucan-release)
1. [The Advanced Way of Theming](#the-advanced-way-of-theming-using-fusionauths-advanced-theme)
1. [The New Visual, No-Code Theme Editor](#the-new-visual-no-code-theme-editor)
1. [Comparing Advanced and Simple Theming](#comparing-advanced-and-simple-theming)
1. [Conclusion](#conclusion)

## Introduction to FusionAuth

FusionAuth is a robust, developer-focused authentication, authorization, and user management platform designed to be seamlessly integrated into any application. Whether you're building a new app from scratch or enhancing an existing system, FusionAuth offers a comprehensive suite of features to handle all aspects of user authentication and management.

Key features of FusionAuth include:

- Single Sign-On (SSO)

> - Multi-Factor Authentication (MFA)

Reviewer: ```suggestion
- [Multi-Factor Authentication (MFA)](/articles/authentication/multi-factor-authentication)
```

- User Registration and Login
- Role-Based Access Control (RBAC)

> - Passwordless Login

Reviewer: ```suggestion
- [Passwordless Login](/features/passwordless)
```

- **Pre-Built Templates**: Access to a variety of professionally designed templates to get you started quickly.
- **Real-Time Preview**: See your changes in real-time as you edit your themes.
- **Customizable Styles**: Fine-tune colors, fonts, and images to match your brand identity.
- **Localization**: Ensure your authentication pages are accessible to all users.

This release aims to empower developers and non-developers alike by providing an intuitive way to create beautiful, functional, and branded authentication experiences.

## The Advanced Way of Theming using FusionAuth's Advanced Theme

Before the introduction of the visual, Simple Theme editor, customizing the look and feel of your FusionAuth login forms required a more traditional approach. This involved editing HTML, CSS, and occasionally JavaScript directly. While our advanced themes are powerful, this method required a certain level of technical expertise and familiarity with web development. If you have not seen this check out our [Themes Overview](/docs/customize/look-and-feel/) page to dive a little deeper.

> You would use our [Themes API](/docs/apis/themes), [FusionAuth CLI](/docs/customize/cli), or by using the Admin UI like below.

Reviewer 2: You would or You could?

> You would use our [Themes API](/docs/apis/themes), [FusionAuth CLI](/docs/customize/cli), or by using the Admin UI like below.

Author: ```suggestion
You could use our [Themes API](/docs/apis/themes), [FusionAuth CLI](/docs/customize/cli), or by using the Admin UI like below.
```

![FusionAuth Admin UI showing Advanced themes](/img/blogs/release-1-51/240530-135039.png)

Advanced Theming in FusionAuth involves the following key components:

- **HTML Templates**: The structure and layout of your authentication pages are defined using HTML templates.
- **CSS Stylesheets**: Custom CSS is used to style these pages according to your branding guidelines.
- **JavaScript**: Optional JavaScript can be included for additional interactivity and functionality.

The Advanced Theming process provides complete control over every aspect of your authentication pages, allowing for highly customized and complex designs. However, it can be time-consuming and challenging, especially for those without a web development background.

You can still edit the Advanced Theme editor from a dropdown within <Breadcrumb>Home / Customizations / Themes </Breadcrumb> like below.

> ![Access Advanced Theme](/img/blogs/release-1-51/240530-134716.png)

Reviewer 2: Image should be cropped.

## The New Visual, No-Code Theme Editor

> The Theming Toucan release (aka 1.51) introduces a game-changing feature: the visual, Simple Theme editor. This tool provides an intuitive, interface that allows you to customize your authentication pages without writing any code. Let's dive into the details of this powerful new feature.

Reviewer 2: ```suggestion
The Theming Toucan release (aka 1.51) introduces a game-changing feature: the visual, Simple Theme editor. This tool provides an intuitive interface that allows you to customize your authentication pages without writing any code. Let's dive into the details of this powerful new feature.
```

- Quickly customize the look and feel of your login forms.
- Use pre-built templates as a starting point.
- Make changes in real-time and see instant previews.
- Customize colors, fonts, and other styles with ease.

### Features and Capabilities

The Simple Theme editor offers a wide range of features to help you create beautiful and functional authentication pages:

1. **Real-Time Preview**: See how your changes will look on the live site as you make them.
1. **Pre-Built Templates**: Start with professionally designed templates and customize them to fit your needs.
1. **Style Customization**: Adjust colors, fonts, backgrounds, and more to match your brand.
1. **Element Properties**: Fine-tune the properties of individual elements, such as buttons and input fields.

### Step-by-Step Guide to Using the Theme Editor

Let's walk through the process of creating a custom theme using the new visual, Simple Theme editor.

#### Step 1: Access the Theme Editor

1. Log in to your FusionAuth administrative dashboard.
2. Navigate to <Breadcrumb>Home / Customizations / Themes </Breadcrumb>
3. Click the **Simple Theme** button to open the theme editor.

> ![Access Simple Theme](/img/blogs/release-1-51/240530-140554.png)

Reviewer 2: Image should be cropped.

#### Step 2: Choose a Template

1. In the theme editor, you'll be presented with a selection of pre-built themes.
2. Browse through the themes and choose one that closely matches your desired layout.

> 3. Click the on Advanced, Dark or Light to start customizing it.

Reviewer 2: ```suggestion
3. Click on Advanced, Dark or Light to start customizing it.
```

![Screen with Preview of Login and Pr-built Theme selections](/img/blogs/release-1-51/240530-140743.png)

#### Step 3: Customize the Images

1. In the right-hand panel, you'll find options to add image paths.

> 1. You can add a Theme logo that will be displayed on top of the login form by adding the image url.

Author: ```suggestion
1. You can add a Theme logo that will be displayed on top of the login form by adding the image URL.
```

> 1. Input a background image url and the appropriate settings.

Author: ```suggestion
1. Input a background image URL and the appropriate settings.
```
